src/main/java/laddergame/model/LadderGenerator.java

```diff
@@ -0,0 +1,19 @@
+package laddergame.model;
+
+import java.util.stream.IntStream;
+import laddergame.model.ladder.Height;
+import laddergame.model.ladder.Ladder;
+import laddergame.model.ladder.Rung;
+import laddergame.model.ladder.RungCreateDecider;
+import laddergame.model.participant.Participants;
+
+public class LadderGenerator {
+
+  public Ladder generateLadder(Participants participants, Height height, RungCreateDecider rungCreateDecider) {
+    return new Ladder(
+        IntStream.range(0, height.getHeight())
+            .mapToObj(index -> new Rung(participants, rungCreateDecider))
+            .toList()
```

Comment:
오호 인트스트림.. 

저는 스트림의 장점을 다음과 같이 봅니다.
1. 컬렉션 내부 요소 수정없이 재활용 가능(원본 요소의 불변성 유지)
2. 복잡한 로직을 함수형으로 표현 가능(가독성 향상)
3. 반복문에 병렬성 쉽게 부여 가능

그런데 지금은 일단 스트림의 요소 값(0에서 사다리 높이까지의 숫자) 은 사용되고 있지 않으니 1번 장점은 없고, 병렬성도 없으니 3번 장점도 없습니다.
그럼 이제 가독성만이 스트림의 사용 이유일 것 같아요. 그런데 밑처럼 일반 포문을 사용하는게 어떤 기능인지가 더 명확해보이지 않나요!?🧐
 
```suggestion
List<Rung> rungs = new ArrayList<>();
for (int i = 0; i < height.getHeight(); i++) {
    rungs.add(new Rung(participants, rungCreateDecider));
}
```

스트림도 생성 비용이나 디버깅등에서 단점이 있기에 사용의 기준을 명확하게 결정하고 사용하는게 좋습니다!
이 기회에 스트림의 장단점 한 번 정리해보시죠👍

Reply:
for문을 쓸까 고민했습니다! 
스트림은 아직 익숙하지 않은 방식이기 때문에 조금 더 써보기 위해 스트림을 사용했습니다.
확실히 이 코드에서 스트림의 장점은 가독성 밖에 없고 이 마저도 for문이 더 명확해보입니다.
스트림은 데이터의 흐름을 처리합니다. 간결하고 직관적인 표현이 가능합니다. 다양한 연산이 있고, 병렬처리가 가능합니다.
원본 데이터를 변경하지 않고 별도로 생성합니다. 따라서 정렬이나 필터링 작업도 별도에서 처리되며 원본 데이터가 변경되지 않습니다.
스트림을 열면 생성 비용이 들기 때문에 무분별한 사용 보다는 상황을 고려하는 것이 좋으며, 순서를 보장해야할 때는 병렬성이 단점이 될 수도 있습니다.